pages/index.tsx

```diff
@@ -147,69 +174,108 @@ export default function Home({
+    );
+
+  if (error) return <p className="mt-5 bg-red-500 text-lg">{error.message}</p>;
 
-                    <SvgTrashCan
-                      key={`${giftItem.uuid}_deletebutton`}
-                      width={24}
-                      height={24}
-                      className="trigger-line-through col-start-2 row-start-1 justify-self-end align-middle text-stone-600 hover:cursor-pointer"
-                      onClick={() => {
-                        setDeleteModalGiftData(giftItem);
-                        setIsDeleteModalOpen(true);
-                      }}
-                    />
-                  </div>
-                </div>
-              ))}
-              {isEditModalOpen && editModalGiftData && (
-                <EditModal
-                  gift={editModalGiftData}
-                  refreshGiftList={() => void refreshGiftList()}
-                  setIsModalOpen={setIsEditModalOpen}
+  return (
+    <div>
+      {giftData.length > 0 && (
+        <div>
+          {giftData.map((giftItem) => (
+            <div
+              key={`${giftItem.uuid}_divbutton`}
+              className="mt-4 animate-opacity"
+            >
+              <div key={giftItem.uuid} className="grid">
+                <p
+                  className={`hover-target col-start-1 text-primaryText [overflow-wrap:anywhere]`}
+                >
+                  {giftItem.gift} <span>-</span> {giftItem.receiver}
+                </p>
+                <SvgPencilEdit
+                  key={`${giftItem.uuid}_editbutton`}
+                  width={24}
+                  height={24}
+                  className="trigger-underline col-start-2 row-start-1 mr-8 justify-self-end align-middle text-stone-600 hover:cursor-pointer"
+                  onClick={() => {
+                    setEditModalGiftData(giftItem);
+                    setIsEditModalOpen(true);
+                  }}
                 />
-              )}
 
-              {isDeleteModalOpen && deleteModalGiftData && (
-                <DeleteModal
-                  gift={deleteModalGiftData}
-                  refreshGiftList={() => void refreshGiftList()}
-                  setIsModalOpen={setIsDeleteModalOpen}
+                <SvgTrashCan
+                  key={`${giftItem.uuid}_deletebutton`}
+                  width={24}
+                  height={24}
+                  className="trigger-line-through col-start-2 row-start-1 justify-self-end align-middle text-stone-600 hover:cursor-pointer"
+                  onClick={() => {
+                    setDeleteModalGiftData(giftItem);
+                    setIsDeleteModalOpen(true);
+                  }}
                 />
-              )}
+              </div>
             </div>
+          ))}
+          {isEditModalOpen && editModalGiftData && (
+            <EditModal
+              gift={editModalGiftData}
+              refreshGiftList={() => void refreshGiftList()}
```

Comment:
Lahjalista päivitetään invalidoimalla query (muutos pitää tehdä muihinkin kohtiin, missä käytetään `refreshGiftList` funktiota).
```suggestion
              refreshGiftList={() => void queryClient.invalidateQueries({ queryKey: ['gifts'] }))}
```